Add a new section on how to collect AWS Network Firewall using Firehose (#3885)
elastic/observability-docs: docs/en/observability/cloud-monitoring/aws/monitor-aws-firewall-firehose.asciidoc

- Create a delivery stream in Amazon Data Firehose
- Set up logging to forward the logs to the Elastic stack using a Firehose stream
- Visualize your logs in {kib}

[discrete]
[[firehose-firewall-prerequisites]]
== Before you begin

We assume that you already have:

- An AWS account with permissions to pull the necessary data from AWS.
- A deployment using our hosted {ess} on {ess-trial}[{ecloud}]. The deployment includes an {es} cluster for storing and searching your data, and {kib} for visualizing and managing your data. AWS Data Firehose works with Elastic Stack version 7.17 or greater, running on Elastic Cloud only.

IMPORTANT: AWS PrivateLink is not supported. Make sure the deployment is on AWS, because the Amazon Data Firehose delivery stream connects specifically to an endpoint that needs to be on AWS.

[discrete]
[[firehose-firewall-step-one]]
== Step 1: Install AWS integration in {kib}

. In {kib}, navigate to *Management* > *Integrations* and browse the catalog to find the AWS integration.

. Navigate to the *Settings* tab and click *Install AWS assets*.

[discrete]
[[firehose-firewall-step-two]]
== Step 2: Select a resource

> [role="screenshot"]

Reviewer: These will look better without our screenshot CSS since they're on a white background
```suggestion
```

image::firehose-networkfirewall-firewall.png[AWS Network Firewall]

You can either use an existing AWS Network Firewall, or create a new one for testing purposes.

Creating a Network Firewall is not trivial and is beyond the scope of this guide. For more information, see the AWS documentation on the https://docs.aws.amazon.com/network-firewall/latest/developerguide/getting-started.html[Getting started with AWS Network Firewall] guide.

> 

Reviewer 2: @dedemorton, explaining how to set up a Network Firewall for testing would be 2x the size of this guide.

So, I guess we have (at least) two options:

- say, "it's beyond the scope of this guide."
- link to an external source 

I created a Terraform module at https://github.com/zmoog/integrations-cookbook/pull/1/files to test the Network Firewall for this guide. The module would allow users to set up the entire Network Firewall in one step. 

However, (1) the module is in draft on (2) my personal repo. For the draft, I can finish it up quickly. But I can't link to my personal repo from the official Elastic docs.

What are our options here? Can we deliver the Terraform module through an official Elastic resource? Any suggestion is welcome!

Reviewer 3: Good question! TBH it's been quite awhile since I've provided users with links to special resources. Does it need to be a GitHub repo? I know in the past we've used download.elastic.co to provide sample data for guides, but this seems different. @bmorelli25 Do you have any ideas for the best way to deliver the Terraform module that Maurizio created for testing the network firewall?

Reviewer: Hi @zmoog. Yes, we have the https://github.com/elastic/observability-examples repo for exactly this purpose.

[discrete]
[[firehose-firewall-step-three]]
== Step 3: Create a stream in Amazon Data Firehose

> [role="screenshot"]

Reviewer: ```suggestion
```

image::firehose-networkfirewall-stream.png[Firehose stream]

. Go to the https://console.aws.amazon.com/[AWS console] and navigate to Amazon Data Firehose.  

. Click *Create Firehose stream* and choose the source and destination of your Firehose stream. Unless you are streaming data from Kinesis Data Streams, set source to `Direct PUT` and destination to `Elastic`. 

. Provide a meaningful *Firehose stream name* that will allow you to identify this delivery stream later. 

NOTE: For advanced use cases, source records can be transformed by invoking a custom Lambda function. When using Elastic integrations, this should not be required.

Set the **Parameters** in the **Destination settings** section.

[discrete]
[[firehose-firewall-step-three-desination-settings-parameters]]
=== Parameters

Use the following parameters:

[cols="1,1",options="header"]
|===
|Name
|Value

| `es_datastream_name`
| `logs-aws.firewall_logs-default`

|===

The Firehose stream is ready to send logs to our Elastic Cloud deployment.

[discrete]
[[firehose-firewall-step-four]]
== Step 4: Enable logging

> [role="screenshot"]

Reviewer: ```suggestion
```

image::firehose-networkfirewall-logging.png[AWS Network Firewall logging]

The AWS Network Firewall logs have built-in logging support. It can send logs to Amazon S3, Amazon CloudWatch, and Amazon Kinesis Data Firehose.

To enable logging to Amazon Data Firehose:

. In the AWS console, navigate to the AWS Network Firewall service.

. Select the firewall for which you want to enable logging.

. In the *Logging* section, click *Edit*.

. Select the *Send logs to* option and choose *Kinesis Data Firehose*.

. Select the Firehose stream you created in the previous step.

. Click *Save*.

At this point, the Network Firewall will start sending logs to the Firehose stream.

[discrete]
[[firehose-firewall-step-five]]
== Step 5: Visualize your Network Firewall logs in {kib}

> With the new logging settings in place, the Network Firewall starts sending log events to the Firehose stream.
> 
> image::firehose-networkfirewall-data-stream.png[Firehose monitor Network Firewall logs]

Reviewer: Move image to the top like in the other sections.

```suggestion
image::firehose-networkfirewall-data-stream.png[Firehose monitor Network Firewall logs]

With the new logging settings in place, the Network Firewall starts sending log events to the Firehose stream.
```

> Navigate to {kib} and choose among the following monitoring options:
> 
> - *Visualize your logs with Discover*

Reviewer: We try to avoid lists of one. How about this?
```suggestion
Navigate to {kib} and choose *Visualize your logs with Discover*.
```

+

> image::firehose-networkfirewall-discover.png[Visualize Network Firewall logs with Discover]

Reviewer: And we should add it here because this one is a screenshot

```suggestion
[role="screenshot"]
image::firehose-networkfirewall-discover.png[Visualize Network Firewall logs with Discover]
```

. Create a VPC and other networking resources.
+
Leave the default settings and choose a name for your VPC resources.
+
[role="screenshot"]
image::firehose-firewall-vpc-resources.png[Firewall VPC resources]

. Create an AWS Network Firewall

. Set up the firewall policy by creating a rule group.

. Deploy an EC2 to generate network traffic.
+
Launch an EC2, select the VPC you have just created, and enable *Auto-assign public IP*.

. Set up logging.
+
Open the *Logging* section to edit your firewall settings. If you want to quickly check your Network Firewall logs before setting up Firehose, you can enable logging on CloudWatch, and then inspect the log events:
+
[role="screenshot"]
image::firehose-firewall-logging.png[Firewall setup logging]

. Visit CloudWatch and open your log group. If everything is working correctly, you will get the list of log events:
+
[role="screenshot"]
image::firehose-cloudwatch-log-events.png[CloudWatch Log events]

[discrete]
[[firehose-firewall-step-three]]
== Step 3: Create a delivery stream in Amazon Data Firehose

. Go to the https://console.aws.amazon.com/[AWS console] and navigate to Amazon Data Firehose.  

> . Click *Create Firehose stream* and choose the source and destination of your Firehose stream. Unless you are streaming data from Kinesis Data Streams, set source to `Direct PUT` and destination to `Elastic`. 
> 
> . Provide a meaningful *Firehose stream name* that will allow you to identify this delivery stream later. Your Firehose name must start with the prefix `aws-waf-logs-` or it will not show up later.
> 
> NOTE: For advanced use cases, source records can be transformed by invoking a custom Lambda function. When using Elastic integrations, this should not be required.
> 

Reviewer 2: Creating a Firehose stream to forward Network Firewall is similar to the CloudTrail step. 

```suggestion
For more information on how to set up a Amazon Data Firehose delivery stream to send data to Elastic Cloud, you can also check the <<monitor-aws-firehose,setup guide>>.

. Click *Create Firehose stream* and choose the source and destination of your Firehose stream. Set source to `Direct PUT` and destination to `Elastic`. 

. Collect {es} endpoint and API key from your deployment on Elastic Cloud.
- Elastic endpoint URL: Enter the Elasticsearch endpoint URL of your Elasticsearch cluster. To find the Elasticsearch endpoint, go to the Elastic Cloud console and select *Connection details*.
- API key: Enter the encoded Elastic API key. To create an API key, go to the Elastic Cloud console, select *Connection details* and click *Create and manage API keys*. If you are using an API key with *Restrict privileges*, make sure to review the Indices privileges to provide at least "auto_configure" & "write" permissions for the indices you will be using with this delivery stream.

. Set up the delivery stream by specifying the following data:
+
- Elastic endpoint URL
- API key
- Content encoding: gzip
- Retry duration: 60 (default) 
- Parameters: 
  - es_datastream_name: `logs-aws.firewall_logs-default`
- Backup settings: failed data only to s3 bucket
```
